### PR TITLE
Support shop logo in svg

### LINKED ISSUE #17673: Allow SVG format for header logo
**Describe the bug**
Currently, we can't use SVG format for header logo.
I think this common format should be allowed.

**To Reproduce**
1. Go to 'Appearance / Theme & logo'
2. Try to upload header logo on SVG format
3. This format is not allowed

## Specs

The .svg extension should be accepted only for header logo
For now we will not allow .svg for email logo because [only 50% of electronic messaging services are supporting it](https://www.caniemail.com/search/?s=svg) and not for the PDF because the system we currently use to generate PDF doesn't support it

https://github.com/PrestaShop/prestashop-specs/blob/master/back-office/design/themes-and-logo/theme-logo.md

### PR DESCRIPTION
**Pull request by jf-viguier**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop 
| Description?  | support shop logo in svg
| Type?         | improvement
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes https://github.com/PrestaShop/PrestaShop/issues/17673
| How to test?  |  add a svg logo

Please help me how to manage the ImageManager::resize($tmpName, _PS_IMG_DIR_ . $logoName) in LogoUpdater::update

It's a non sense to resize and svg, it's vector graphic.
I don't know if resize in used in other format (jpg, png, ...)

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/21627)

### CODE REVIEW

**Comment by jf-viguier** [comment]
I've finished it. I copy file instead of resize and use .svg extension.
@PululuK please remove wip tag.

**Comment by Progi1984** [comment]
@PrestaShop/product-team Need answer in #17673

**Comment by jf-viguier** [comment]
@Progi1984 All checks ok now. I think that a modern cms should support svg natively : shop logos are cleaner and lighter in vector format, what do you think ?
Do we really need product-team to decide on this ?

**Comment by Mch0** [comment]
Nice idea. Very usefull 👍 . Thx

**Comment by ghost** [comment]
Good for the new devices ;)

File: src/Core/Shop/LogoUploader.php [hunk 1]
```diff
@@ -97,7 +98,7 @@ public function update($fieldName, $logoPrefix, array $files = [])
         $files = empty($files) ? $_FILES : $files;
 
         if (isset($files[$fieldName]['tmp_name'], $files[$fieldName]['tmp_name'], $files[$fieldName]['size'])) {
-            if ($error = ImageManager::validateUpload($files[$fieldName], Tools::getMaxUploadSize())) {
+            if ($error = ImageManager::validateUpload($files[$fieldName], Tools::getMaxUploadSize(), ShopLogoSettings::AVAILABLE_LOGO_IMAGE_EXTENSIONS)) {
```
Comment (matks) [inline src/Core/Shop/LogoUploader.php]: Mmmm 🤔 in theory DTOs are used to carry some data around. It feels weird to use it in this LogoUploader class.

Maybe we should extract constant AVAILABLE_LOGO_IMAGE_EXTENSIONS outside of the DTO in a more generic / reusable class ? And DTO would use it too.

**Review by matks — COMMENTED** [review]
I dont know well this area of the code but I see no defects in this code.

However I think what we need to do is think about all the possible usages of logos and try to check whether this will fit well with it 🤔 .

For example
- where is this logo displayed ? are there locations where using SVG might create issues ? on invoices for example ? on data export ? or when used through meta tags such as Twitter previews ?
- does SVG format fit well with how logos are being managed in prestashop ? as you mentioned, it does not make sense to perform resize operations on it, what about others operations ?

**Comment by jf-viguier** [comment]
Can I have a validation for hacktoberfest ?

**Comment by jf-viguier** [comment]
> where is this logo displayed ? 
If no logo is uploaded in email/invoice logo box, the header logo is used here and it's an issue @matks 
- It's ok for svg in email (thunderbird, not ok in gmail)
- It's not working in pdf invoice, can someone help on this ?

**Comment by jf-viguier** [comment]
@LouiseBonnard I've done a PR here https://github.com/PrestaShop/PrestaShop/pull/21627
For shop logo. I need help pour pdf generating.

**Comment by LouiseBonnard** [comment]
@jf-viguier, yep, thanks a lot! I think @Progi1984 will be able to help you with the PDF support. ;-)

As for the shop logo, just to be sure, you are talking about the header logo or the email/invoice logo section of the Design > Theme & Logo page? I'm not very sure about the second section, you'll tell me about it:

1. Header logo is meant to be displayed in the front office. In other words, it is related to the browser and, according to https://caniuse.com/?search=svg, the support coverage reaches very satisfying percentages.

2. PDF should work fine in most cases, as stated by @Progi1984.

2. On the contrary, in most cases, sending SVG images in emails might not work if you look at https://www.caniemail.com/search/?s=svg. It shows that the support coverage is quite poor. Maybe we should wait a bit before implementing SVG support for the email logo?

**Comment by jf-viguier** [comment]
@LouiseBonnard my pr is for the header logo on the website but if no email/invoice logo is set, the header logo is used : here is the issue. 
> PDF should work fine in most cases, as stated by @Progi1984
where can I find this information ?

**Comment by LouiseBonnard** [comment]
Nice, it gets tricky, aha! I can see two solutions: 

- either we allow it and notify the users with the tooltip or any notification;
- or we simply do not allow the SVG image format for any of the shop logos.

I'll let @Progi1984 get back to us regarding the PDF, I don't have all the details. :-(

**Comment by jf-viguier** [comment]
@LouiseBonnard @Progi1984 any news here ? For PDF ? 
tooltip is ok for me

**Comment by marionf** [comment]
Hello everyone,

First the .svg extension should be accepted only for header logo
For now we will not allow .svg for email logo because [only 50% of electronic messaging services are supporting it](https://www.caniemail.com/search/?s=svg) and not for the PD files because the system we currently use to generate PDF doesn't support it

@jf-viguier Could you please resolve the conflicting file ? Then, we will be able to review the code of your PR :)

**Comment by jf-viguier** [comment]
@marionf conflict resolved

**Comment by Progi1984** [comment]
@jf-viguier Could you rebase and remove merge commits in your PR ?

**Review by Progi1984 — DISMISSED** [review]

**Comment by jf-viguier** [comment]
@Progi1984 I've done a rebase but I need help to remove merge commit please.

**Comment by jf-viguier** [comment]
@ui test PHP 7.4 failed but it's not related to my fix, how to relauch ?

**Comment by jf-viguier** [comment]
I had a conflict issue and @Progi1984 deleted my commits ;-)
I've made a new PR, please validate @matks @Progi1984, it's the same code here :
https://github.com/PrestaShop/PrestaShop/pull/23959

It's dead for hacktoberfest ;-)

**Comment by Progi1984** [comment]
Error in rebase, sorry :p

**Pull request closed without merging**